twilight-model/src/application/interaction/application_command/option.rs

```diff
@@ -202,8 +202,10 @@ impl<'de> Deserialize<'de> for CommandDataOption {
                         CommandOptionType::Attachment => {
                             let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
 
-                            if let ValueEnvelope::Id(id) = val {
-                                CommandOptionValue::Attachment(id.cast())
+                            if let ValueEnvelope::String(id) = &val {
+                                CommandOptionValue::Attachment(id.parse().map_err(|_e| {
```

Comment:
Just for the sake of consistency

```suggestion
                                CommandOptionValue::Attachment(id.parse().map_err(|_| {
```